src/linear_algebra/bilinear_form.lean

```diff
@@ -400,3 +410,175 @@ begin
 end
 
 end alt_bilin_form
+
+namespace bilin_form
+
+section linear_adjoints
+
+variables {R : Type u} [comm_ring R]
+variables {M : Type v} [add_comm_group M] [module R M]
+variables {M₂ : Type v} [add_comm_group M₂] [module R M₂]
+variables (B B' : bilin_form R M) (B₂ : bilin_form R M₂)
```

Comment:
I think I would make all of these implicit.

Reply:
Very brief experiments with this seem to be a little troublesome. I may just have to select a couple of judicious locations to say `variables (B B' B₂)` and `variables {B B' B₂}`.

I hope to have a few minutes later this evening to try again.

Reply:
Now done.